src/OrleansCodeGenerator/GrainMethodInvokerGenerator.cs

```diff
@@ -187,31 +187,56 @@ private static MethodDeclarationSyntax GenerateExtensionInvokeMethod(Type grainT
         /// </returns>
         private static MethodDeclarationSyntax GenerateInvokeMethod(Type grainType, MethodInfo invokeMethod)
         {
-            var methodDeclaration = invokeMethod.GetDeclarationSyntax();
             var parameters = invokeMethod.GetParameters();
 
             var grainArgument = parameters[0].Name.ToIdentifierName();
-            var interfaceIdArgument = parameters[1].Name.ToIdentifierName();
-            var methodIdArgument = parameters[2].Name.ToIdentifierName();
-            var argumentsArgument = parameters[3].Name.ToIdentifierName();
+            var requestArgument = parameters[1].Name.ToIdentifierName();
+            
+            // Store the relevant values from the request in local variables.
+            var interfaceIdDeclaration =
```

Comment:
In general, all the `var`s in this new code need to be changed to specific types for clarity.

DotNet GitHub coding style is to only use `var` with `new` typed object constructor statements - see Item 10 
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md